arrow-select/src/take.rs

```diff
@@ -1985,6 +1987,23 @@ mod tests {
         assert_eq!(&values, &[Some(23), Some(4), None, None])
     }
 
+    #[test]
+    fn test_take_fixed_size_list_null_indices() {
+        let indices = Int32Array::new(vec![0, 1].into(), Some(NullBuffer::from(vec![true, false])));
+        let values = Arc::new(Int32Array::from(vec![0, 1, 2, 3]));
+        let arr_field = Arc::new(Field::new("item", values.data_type().clone(), true));
+        let values = FixedSizeListArray::try_new(arr_field, 2, values, None).unwrap();
+
+        let r = take(&values, &indices, None).unwrap();
+        let values = r
+            .as_fixed_size_list()
+            .values()
+            .as_primitive::<Int32Type>()
+            .into_iter()
+            .collect::<Vec<_>>();
+        assert_eq!(&values, &[Some(0), Some(1), None, None])
```

Comment:
nit: You can directly compare `&ScalarBuffer` and a slice.
```suggestion
        let values = r
            .as_fixed_size_list()
            .values()
            .as_primitive::<Int32Type>()
            .values();
        assert_eq!(values, &[Some(0), Some(1), None, None])
```

Reply:
You can compare `ScalarBuffer` with `[T]` but not with `[Option<T>]`